### PR TITLE
feat: add feature to record multi-host system

### PR DESCRIPTION
**Pull request by ymski**
## Description

This PR adds feature to record multi-host system.

## Related links

- https://github.com/tier4/caret_analyze/pull/455
- https://github.com/tier4/caret_trace/pull/262
- https://github.com/tier4/caret_doc/pull/190

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been properly tested.
- [x] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: ros2caret/verb/record.py [hunk 1]
```diff
@@ -180,6 +180,14 @@ def main(self, *, args):
         context_names = names.DEFAULT_CONTEXT
         events_kernel = []
 
+        # check lttng session exist
+        command = "lttng list | grep --quiet \'\\[active\\]\'"
+        ps = subprocess.run(command, shell=True, capture_output=True, text=True)
```
Comment (isp-uetsuki) [inline ros2caret/verb/record.py:185]: Are `capture_output=True` and `text=True` necessary? If they are not, I suggest removing them.

**Review by isp-uetsuki — CHANGES_REQUESTED** [review]

File: ros2caret/verb/record.py (hunk 1 above)
Comment (ymski) [reply ros2caret/verb/record.py]: remove options in [ad468ea](https://github.com/tier4/ros2caret/pull/163/commits/ad468eada1a706aa4d9f962b9c838a0d00da3f1a).

File: ros2caret/verb/chain_latency.py [hunk 2]
```diff
@@ -43,7 +43,8 @@ def add_arguments(self, parser, cli_name):
             help='granularity of trace points to be visualized')
 
     def main(self, *, args):
-        lttng = Lttng(args.trace_directory, force_conversion=True)
+        lttng = Lttng(args.trace_directory, force_conversion=True,
+                      validate=isinstance(args.trace_directory, list))
```
[the remark this answers is not in the document]
Comment (ymski) [reply ros2caret/verb/chain_latency.py]: @isp-uetsuki 
Thanks for reviewing.
Referring to the following document, it appears that `args.trace_directory` is a list type even when a single directory is passed to the -t option.
- https://docs.python.org/3/library/argparse.html#nargs

I have actually run it and confirmed that it works as expected.

[the remark this answers is not in the document]
Comment (isp-uetsuki) [reply ros2caret/verb/chain_latency.py]: Can you also apply following changes?

> - please change message_flow.py and summary.py as the same manner.
> - please change type of trace_dir argument of CreateArchitecture.__init__ to list[str].

**Comment by isp-uetsuki** [comment]
LGTM

**Review by isp-uetsuki — APPROVED** [review]
LGTM

**Review by xygyo77 — APPROVED** [review]
LGTM

**Pull request merged**